crates/sui-cluster-test/src/faucet.rs

```diff
@@ -114,34 +115,34 @@ impl FaucetClient for RemoteFaucetClient {
     }
 }
 
-/// A dummy faucet that does nothing, suitable for local cluster testing
-/// where gas coins are prepared in genesis
-pub struct DummyFaucetClient {}
+/// A local faucet that holds some coins since genesis
+pub struct LocalFaucetClient {
+    simple_faucet: SimpleFaucet,
+}
 
-impl DummyFaucetClient {
-    fn new() -> Self {
-        info!("Use dummy faucet");
-        Self {}
+impl LocalFaucetClient {
+    fn new(simple_faucet: SimpleFaucet) -> Self {
+        info!("Use local faucet");
+        Self { simple_faucet }
     }
 }
 #[async_trait]
-impl FaucetClient for DummyFaucetClient {
-    /// Dummy faucet client does not request coins from a real faucet.
-    /// Instead it just syncs all gas objects for the address.
+impl FaucetClient for LocalFaucetClient {
     async fn request_sui_coins(
         &self,
         client: &WalletClient,
         minimum_coins: Option<usize>,
     ) -> Result<Vec<GasCoin>, anyhow::Error> {
-        let wallet = client.get_wallet();
         let address = client.get_wallet_address();
-        client.sync_account_state().await?;
-        let gas_coins = wallet
-            .gas_objects(address)
-            .await?
-            .iter()
-            .map(|(_amount, o)| GasCoin::try_from(o).unwrap())
-            .collect::<Vec<_>>();
+        let receipt = self
+            .simple_faucet
+            .send(Uuid::new_v4(), address, &[50000; 5])
+            .await
+            .unwrap_or_else(|err| panic!("Failed to get gas tokens with error: {}", err));
+
+        sleep(Duration::from_secs(2)).await;
```

Comment:
just curious - why is this sleep needed?

Reply:
this is to let FN to catch up with the txns. Ideally when we have txn subscription from FN we should move away from this

Reply:
Can we leave this as a comment on the line above sleep?